icu4j/main/tests/core/src/com/ibm/icu/dev/test/impl/UnitsTest.java

```diff
@@ -40,79 +40,126 @@ public static boolean compareTwoBigDecimal(BigDecimal expected, BigDecimal actua
         output = MeasureUnit.forIdentifier("light-year-and-meter");
         final MeasureUnitImpl inputImpl3 = MeasureUnitImpl.forIdentifier(input.getIdentifier());
         final MeasureUnitImpl outputImpl3 = MeasureUnitImpl.forIdentifier(output.getIdentifier());
-
         ComplexUnitsConverter converter3 = new ComplexUnitsConverter(inputImpl3, outputImpl3, rates);
 
-        List<Measure> measures3 =
-            converter3.convert(BigDecimal.valueOf(2.0).subtract(ComplexUnitsConverter.EPSILON), null);
-        assertEquals("measures length", 2, measures3.size());
-        assertEquals("light-year test: measures[0] value", BigDecimal.valueOf(2), measures3.get(0).getNumber());
-        assertEquals("light-year test: measures[0] unit", MeasureUnit.LIGHT_YEAR.getIdentifier(),
-                measures3.get(0).getUnit().getIdentifier());
-        assertEquals("light-year test: measures[1] value", BigDecimal.ZERO, measures3.get(1).getNumber());
-        assertEquals("light-year test: measures[1] unit", MeasureUnit.METER.getIdentifier(),
-                measures3.get(1).getUnit().getIdentifier());
-
-        // 1e-15 light years is 9.46073 meters (calculated using "bc" and the CLDR
-        // conversion factor). With double-precision maths, we get 10.5. In this
-        // case, we're off by almost 1 meter.
-        List<Measure> measures4 = converter3.convert(BigDecimal.valueOf(1.0 + 1e-15), null);
-        assertEquals("measures length", 2, measures4.size());
-        assertEquals("light-year test: measures[0] value", BigDecimal.ONE, measures4.get(0).getNumber());
-        assertEquals("light-year test: measures[0] unit", MeasureUnit.LIGHT_YEAR.getIdentifier(),
-                measures4.get(0).getUnit().getIdentifier());
-        assertTrue("light-year test: measures[1] value", compareTwoBigDecimal(BigDecimal.valueOf(10),
-                BigDecimal.valueOf(measures4.get(1).getNumber().doubleValue()),
-                BigDecimal.valueOf(1)));
-        assertEquals("light-year test: measures[1] unit", MeasureUnit.METER.getIdentifier(),
-                measures4.get(1).getUnit().getIdentifier());
-
-        // 2e-16 light years is 1.892146 meters. We consider this in the noise, and
-        // thus expect a 0. (This test fails when 2e-16 is increased to 4e-16.)
+        // TODO(icu-units#108): the test below is simply broken. Dates back to
+        // the first version of this test case. "1.0 + 2e-17" cannot be
+        // represented by an IEEE 754 double, because 2e-17 is smaller than
+        // 2^-53. (There are 53 bits in the significand.) And the comment still
+        // says "2e-16". Fix as part of #108 and fold into the "cases" above.
+        //
+        // 2e-16 light years is 1.892146 meters. We consider this in the noise,
+        // and thus expect a 0. (This test fails when 2e-16 is increased to
+        // 4e-16.)
+        assertEquals("FIXME: demonstrating brokenness", 1.0 + 2e-17, 1.0, 0);
```

Comment:
I've changed this test case as I moved it into the testCases array, because value of "1.0 + 2e-17" is just "1.0". (2e-17 is smaller than what can be represented by a double with exponent 0.)